compiler/src/iree/compiler/Codegen/Common/GPUDistributeSharedMemoryCopy.cpp

```diff
@@ -158,6 +167,12 @@ static void distributeTransferRead(func::FuncOp funcOp, Value flatThreadId,
     Value id = flatThreadId;
     SmallVector<int64_t, 2> multiplier;
     auto shape = readOp.getVectorType().getShape();
+    int targetVectorSize = 4;
```

Comment:
Maybe this needs to be moved into a utility function.

Reply:
Makes sense, @ThomasRaoux  also suggested making this choice of 4 for fp32 and 8 for fp16 in one place only, I will discuss how best to do that with him tomorrow.